### PR TITLE
feat: use separate worker to build and publish images

### PR DESCRIPTION
**Pull request by mndeveci**
Running image build & publish in parallel;
- Create `Makefile` targets to build each runtime individually
- Create pytest markers to group tests by runtime

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by wchengru — APPROVED** [review]

File: pytest.ini [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[pytest]
```
Comment (hoffa) [inline pytest.ini:1]: I don't think this file is actually needed, markers should work automatically.

File: Makefile [hunk 2]
```diff
@@ -1,14 +1,57 @@
...
+else
+	@echo "Building runtime $(RUNTIME)"
+endif
+
+build-single-arch: pre-build
+	docker build -f build-image-src/Dockerfile-$(RUNTIME) -t amazon/aws-sam-cli-build-image-$(IS_$(RUNTIME)):x86_64 --build-arg SAM_CLI_VERSION=$(SAM_CLI_VERSION) ./build-image-src
+
+build-multi-arch: pre-build
+	docker build -f build-image-src/Dockerfile-$(RUNTIME) -t amazon/aws-sam-cli-build-image-$(IS_$(RUNTIME)):x86_64 --platform linux/amd64 --build-arg SAM_CLI_VERSION=$(SAM_CLI_VERSION) --build-arg AWS_CLI_ARCH=x86_64 --build-arg IMAGE_ARCH=x86_64 ./build-image-src
+	docker rmi public.ecr.aws/amazonlinux/amazonlinux:2
```
Comment (hoffa) [inline Makefile:49]: Why is this needed?

File: pytest.ini (hunk 1 above)
Comment (mndeveci) [reply pytest.ini:1]: You are right, but it gets rid of warning messages in the end of the test run.

File: Makefile (hunk 2 above)
Comment (mndeveci) [reply Makefile:49]: This is required when building a runtime for both X86 and arm64. Without this line, arm64 (the later one) will have contents from X86 image, which cause build to fail. We have same line in here as well; https://github.com/aws/aws-sam-build-images/blob/develop/build-image-src/build_all_images.sh#L44

**Review by hoffa — APPROVED** [review]
:shipit:
